mozilla/remote-settings
# Browser-tests and integration-tests merge, paving trail for better gatekeeper tests

Resolves #565 

Phase 1: (done, see [this commit](https://github.com/mozilla/remote-settings/pull/566/commits/67b7ee77b3a1b5f7fd8bf94da65e2081ac928257))
- Run integration-tests and browser tests all together
- Perform all tests through playwright or kinto clients
- Update documentation

Phase 2: (done, see [this commit](https://github.com/mozilla/remote-settings/pull/566/commits/fbe068d02e77440ce97d1ec4bfcdabafe8a6f7dd))
 - Simplify test parameters
 - No longer allowed to skip test setup steps
 - Update documentation
 - Cleanup a bit

Phase 3: (to follow-up on)
 - Identify personas and their most common workflows
 - Expand tests to cover those

Phase 4: (to follow-up on)
 - Implement automated tests against stage (scheduled or after stage deployment)

Phase 5: (dreaming)
 - Automated releases to production


Open questions:

- Did I simplify the test parameters enough? 
  - We're now able to define test users, but can no longer define parameters that wildly change test behavior.
  - I'm hoping this makes testing against any environments very easy in the future via environment vars
- Did I miss any documentation that needs to be updated?
- Do we like/dislike any patterns in particular in here?

Post merge ToDo's:
- #569
